install/cupy_builder/_compiler.py

```diff
@@ -94,12 +94,18 @@ def _nvcc_gencode_options(cuda_version: int) -> list[str]:
                          ('compute_86', 'sm_86'),
                          ('compute_89', 'sm_89'),
                          ('compute_90', 'sm_90'),]
-            if cuda_version >= 12080:
+            if cuda_version < 12080:
+                arch_list.append('compute_90')
+            elif 12080 <= cuda_version < 12090:
                 arch_list += [('compute_100', 'sm_100'),
                               ('compute_120', 'sm_120'),
                               'compute_100']
-            else:
-                arch_list.append('compute_90')
+            elif 12090 <= cuda_version:
+                arch_list += [('compute_100f', 'sm_100'),
+                              ('compute_103f', 'sm_103'),
```

Comment:
I think we can drop the 103f line and let 100f kick in (by design of the family-specific arch):
https://developer.nvidia.com/blog/nvidia-blackwell-and-nvidia-cuda-12-9-introduce-family-specific-architecture-features/#family-specific_feature_set_introduced_in_nvidia_blackwell
```suggestion
                arch_list += [('compute_100f', 'sm_100'),
```